test/unit/connectors/EnhancedApiNotificationQueueConnectorSpec.scala

```diff
@@ -113,5 +134,25 @@ class EnhancedApiNotificationQueueConnectorSpec extends UnitSpec with MockitoSug
       result.left.get.message shouldBe "unauthorised exception"
       result.left.get.responseCode shouldBe 500
     }
+
+    "return an empty list of unpulled notifications when the downstream returns an empty list" in new Setup {
+
+      when(mockHttpClient.GET[Notifications](meq(s"http://api-notification-queue.url/notifications/unpulled"))
+        (any[HttpReads[Notifications]](), any[HeaderCarrier](), any[ExecutionContext])).thenReturn(Future.successful(Notifications(List())))
+
+      val result = await(enhancedApiNotificationQueueConnector.getAllNotificationsBy(Unpulled))
+
+      result shouldBe Notifications(List())
+    }
+
```

Comment:
I think this should be `return an empty list of previously pulled notifications when the downstream returns an empty list`. Also other tests names that mention `pulled` should follow suit.

Reply:
:+1: